src/utils/application.data.ts

```diff
@@ -76,6 +76,16 @@ export const checkMOsDetailsEntered = (appData: ApplicationData): boolean => {
   return Boolean( appData[ManagingOfficerKey]?.length || appData[ManagingOfficerCorporateKey]?.length ) ;
 };
 
+export const hasNotAddedOrCeasedBos = (appData: ApplicationData): boolean => {
+  if (allBeneficialOwners(appData).some((bo) => !bo.ch_reference)) {
```

Comment:
I think this needs tweaked slightly.

If we've no ceased BOs this will return false prematurely, as we may still have added new BOs that have not ceased.

Could the two conditions be combined too? 

```javascript
const hasAddedOrCeasedABO = allBeneficialOwners(appData).some(bo => !bo.ch_reference || (bo.ceased_date && Object.keys(bo.ceased_date).length !== 0));

return !hasAddedOrCeasedABO;
```

(Separated the return statement to negate the result for readability, because the name of the method is the opposite of what the actual check is looking for. I'd almost argue to rename the method and negate the result where it's used but I'll not :joy:).

Reply:
So currently all we need to check for is that no bos have been added or ceased. So if either a BO has been added this should be false regardless.

I have taken the other comments from this in and added together and negated

Reply:
Ah, cool :thumbsup: The order of these checks has changed since I started writing that comment; I hope it was originally checking the ceased date first :joy:

Reply:
Yep 😄